src/xss-filters.js

```diff
@@ -251,6 +251,57 @@ exports._getPrivFilters = function () {
         }
         return null;
     }
+
+    /* 
+     * Create URL whitelist filter
+     * Ref: https://url.spec.whatwg.org/#url-parsing
+     * @param {boolean} allowAllSafeURIs - to enable inherited protocol (//), blob:, http://, https://, ftp://, "data:image/[gif,jpeg,png];base64," are allowed
+     * @param {boolean} allowInheritProto - to enable inherited protocol (//), or otherwise, only http:// and https:// are allowed
+     * @param {boolean} allowRelPath - to allow relative path
+     * @param {boolean} allowHosts - an optional array of hostnames that each matches /^[\w\.-]+$/. If any one is found unmatched, return null
+     * @param {boolean} reqHtmlDecode - to run html decoding first before applying the tests
+     * @returns {function|null} the yuwl filter that tests value according to the config
+     */
+    function yuwlFactory (allowAllSafeURIs, allowInheritProto, allowRelPath, allowHosts, reqHtmlDecode) {
+        var i, n, reHost, reHosts,
+            reProto = allowAllSafeURIs ? 
+                        /^[\x00-\x20]*(?:(?:(blob:)?https?|ftp):\/\/|data:image\/(?:gif|jpe?g|png);base64,|\/\/)/i :
+                        allowInheritProto ? 
+                            /^[\x00-\x20]*(?:https?:\/\/|\/\/)/i :
+                            /^[\x00-\x20]*https?:\/\//i,
+            reRelPath = allowRelPath && /^[\x00-\x20]*(?![a-z][a-z0-9+-.]*:|[\/\\]{2})/i;
+
+        // create reHosts from the hostList array, that is case insensitive
```

Comment:
okay, so when a boolean flag is enabled to allowSubdomain, and that the allowHosts contains `yahoo.com`. it will pass no matter the levels are, right? therefore, `a.yahoo.com` and `a.b.yahoo.com` will both pass.